internal/gatewayapi/listener.go

```diff
@@ -145,12 +146,8 @@ func (t *Translator) processInfraIRListener(listener *ListenerContext, infraIR I
 		proto = ir.UDPProtocolType
 	}
 
-	infraPortName := string(listener.Name)
-	if t.MergeGateways {
-		infraPortName = irHTTPListenerName(listener)
-	}
 	infraPort := ir.ListenerPort{
-		Name:          infraPortName,
+		Name:          strings.ToLower(fmt.Sprintf("%s-%d", proto, servicePort.port)),
```

Comment:
what about move to a helper/util packge?

Reply:
moved into a helper pkg